app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt

```diff
@@ -0,0 +1,324 @@
+  val buttonHeight = screenHeight * buttonHeightFactor
+  val buttonTextSize = (buttonHeight.value * textSizeFactor).sp
+
+  var manSelected by remember { mutableStateOf(false) }
+  var womanSelected by remember { mutableStateOf(false) }
+  var otherSelected by remember { mutableStateOf(false) }
+  var ageFrom by remember { mutableStateOf<Int?>(null) }
+  var ageTo by remember { mutableStateOf<Int?>(null) }
+
+  // test values
+  var selectedTags = remember {
+    mutableStateOf<List<Pair<String, Color>>>(listOf(Pair("salsa", Color.Red)))
+  }
+  val availableTags = remember {
+    mutableStateOf<List<Pair<String, Color>>>(
+        listOf(
+            Pair("salsa", Color.Red),
+            Pair("pesto", Color.Green),
+            Pair("psto", Color.Green),
+            Pair("pest", Color.Green),
+            Pair("peest", Color.Green)))
+  }
+
+  val stringQuery = remember { mutableStateOf("") }
+  val expanded = remember { mutableStateOf(false) }
+
+  var ageFromText by remember { mutableStateOf("") }
+  var ageToText by remember { mutableStateOf("") }
+  var ageRangeError by remember { mutableStateOf(false) }
+
+  fun validateAndUpdateAgeFrom(input: String) {
+    ageFromText = input
+    if (input.isEmpty()) {
+      ageFrom = null
+      ageRangeError = false
+    } else {
+      val newAge = input.toIntOrNull()
+      if (newAge != null && newAge >= 13) {
+        ageFrom = newAge
+        ageRangeError = ageTo != null && newAge > ageTo!!
+      } else {
+        ageFrom = null
+        ageRangeError = false
+      }
+    }
+  }
+
+  fun validateAndUpdateAgeTo(input: String) {
+    ageToText = input
+    if (input.isEmpty()) {
+      ageTo = null
+      ageRangeError = false
+    } else {
+      val newAge = input.toIntOrNull()
+      if (newAge != null && newAge >= 13) {
+        ageTo = newAge
+        ageRangeError = ageFrom != null && newAge < ageFrom!!
+      } else {
+        ageTo = null
+        ageRangeError = false
+      }
+    }
+  }
+
+  fun validateAgeRange() {
+    when {
+      ageRangeError -> {
+        Toast.makeText(context, "Invalid age range", Toast.LENGTH_SHORT).show()
+        ageToText = ""
+        ageFromText = ""
+        return // exit early so that there aren't 3 overlapping toasts as the next 2 would also be
+        // called
+      }
+      ageFromText.isNotEmpty() && ageFrom == null -> {
+        Toast.makeText(context, "Please enter a valid 'From' age (13 or older)", Toast.LENGTH_SHORT)
+            .show()
+        ageFromText = "" // Clear invalid input
+      }
+      ageToText.isNotEmpty() && ageTo == null -> {
+        Toast.makeText(context, "Please enter a valid 'To' age (13 or older)", Toast.LENGTH_SHORT)
+            .show()
+        ageToText = "" // Clear invalid input
+      }
+    }
+  }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Text("Filter", color = Color.White, modifier = Modifier.testTag("FilterTopBarTitle"))
```

Comment:
Would be more readable to have a composable funciton outside of the Scaffold that does the TopAppBar ? It is ok to keep it like this right now but Since we use the same TopAppBar for the Filter, the Notifications, the AroundYou and the Settings, we put a standardized TopAppBar later on

Reply:
This doesn't need to be fixed, it is just a remark

Reply:
You're right, it's temporary.